hid-ghlive/src/hid-ghlive.c

```diff
@@ -220,8 +222,23 @@ static int ghlive_probe(struct hid_device *hdev,
 		timer_setup(&sc->poke_timer, ghl_magic_poke_ps3wiiu, 0);
 	}
 	else if (sc->quirks & GHL_GUITAR_PS4){
+		
+		struct usb_interface *intf = to_usb_interface(sc->hdev->dev.parent);
+
+		if (intf->cur_altsetting->desc.bNumEndpoints != 2)
+		return -ENODEV;
+				
+		for (i = 0; i < 2; i++) {
+			struct usb_endpoint_descriptor *ep =
+					&intf->cur_altsetting->endpoint[i].desc;
+	
+			if (usb_endpoint_xfer_int(ep)) {
+				if (!usb_endpoint_dir_in(ep))
```

Comment:
Now that I am looking at it more closely, i could use usb_endpoint_dir_out() instead of !usb_endpoitn_dir_in(). Again, i was using the xpad.c code where they used both the IN and OUT endpoints, but since i need the OUT only, i could change this. Should I?

Reply:
Agreed. Please make the change.